src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php

```diff
@@ -234,15 +234,27 @@ public function testEncodeRootAttributes()
     public function testEncodeCdataWrapping()
     {
         $array = [
-            'firstname' => 'Paul & Martha <or Me>',
+            'firstname' => 'Paul & Martha O\'Donnel <or Me>',
         ];
 
         $expected = '<?xml version="1.0"?>'."\n".
-            '<response><firstname><![CDATA[Paul & Martha <or Me>]]></firstname></response>'."\n";
+            '<response><firstname><![CDATA[Paul & Martha O\'Donnel <or Me>]]></firstname></response>'."\n";
 
         $this->assertEquals($expected, $this->encoder->encode($array, 'xml'));
     }
 
+    public function testEncodeCdataWrappingWithCustomPattern()
+    {
+        $array = [
+            'lastname' => 'O\'Donnel',
+        ];
+
+        $expectedWrapped = '<?xml version="1.0"?>'."\n".
+            '<response><lastname><![CDATA[O\'Donnel]]></lastname></response>'."\n";
```

Comment:
Why does both tests produce the same result, even if the wrapping patterns are different?

`/[<>&]/` <> `/[<>&"\']/`

Reply:
I didn't think about it but the change you asked me to do [here](https://github.com/symfony/symfony/pull/54663#discussion_r1572763664) it's indeed not covering the case I was aiming to cover with the previous submitted version.
In my original test I wanted to check that the default pattern was not considering the single quote as a candidate for a CDATA wrapping and that it was considering when including it with a custom regex.
The `Paul & Martha <or Me>` test already includes characters that trigger a CDATA wrapping so it does not make any difference.
Shall I revert to my original proposition or do you prefer that I create a separate test case?

Reply:
Yes please revert it then, thanks

Reply:
Done

Reply:
It's in the same test, the first assertion verifies that the `O'Donnel` value is not wrapped in a CDATA structure when using the default pattern while the second assertion verifies that, including the single quote in the pattern, results in `O'Donnel` value wrapped.
IMHO it's part of the same test case, that's why I wrote it this way.

If you prefer I could structure it differently if you prefer by doing:
```
testEncodeCdataWrappingWithDefaultPattern
  input: Paul & Martha <or Me>
  pattern: default
  expect: wraps in CDATA structure  

testDoesNotEncodeCdataWrappingWithDefaultPattern
  input: O'Donnel
  pattern: default
  expect: does not wrap in CDATA structure  

testEncodeCdataWrappingWithCustomPattern
  input: O'Donnel
  pattern: custom
  expect: wraps in CDATA structure  
```

Reply:
Ah right, von we make to testcases instead of one with 2 assertions? Or a dataprovider

Reply:
With data providers I would split the test cases in two, one for the default pattern and one for the custom pattern.
Would something like this work for you?
```
testEncodeCdataWrappingWithDefaultPattern(array $input, string $expected)
  case 1 ('Paul & Martha', 'not wrapped')
  case 2 ('O\'Donnel', 'not wrapped')
  case 3 ('Paul & Martha <or Me>', 'wrapped')

testEncodeCdataWrappingWithCustomPattern(array $input, string $expected)
  case 1 ('Paul & Martha', 'not wrapped')
  case 2 ('O\'Donnel', 'wrapped')
  case 3 ('Paul & Martha <or Me>', 'wrapped')
```

Reply:
Yes 🙌

Reply:
Done, thanks for your feedback!